rust/kernel/file_operations.rs

```diff
@@ -356,26 +356,12 @@ pub struct IoctlCommand {
 
 impl IoctlCommand {
     /// Constructs a new [`IoctlCommand`].
-    ///
-    /// # Safety
-    ///
-    /// The caller must ensure that `fs` is compatible with `arg` and the original caller's
-    /// context. For example, if the original caller is from userland (e.g., through the ioctl
-    /// syscall), then `arg` is untrusted and `fs` should therefore be `USER_DS`.
-    unsafe fn new(cmd: u32, arg: usize) -> Self {
-        let user_slice = {
-            let dir = (cmd >> bindings::_IOC_DIRSHIFT) & bindings::_IOC_DIRMASK;
-            if dir == bindings::_IOC_NONE {
-                None
-            } else {
-                let size = (cmd >> bindings::_IOC_SIZESHIFT) & bindings::_IOC_SIZEMASK;
-
-                // SAFETY: We only create one instance of the user slice, so TOCTOU issues are not
-                // possible. The `set_fs` requirements are imposed on the caller.
-                UserSlicePtr::new(arg as _, size as _).ok()
-            }
-        };
+    fn new(cmd: u32, arg: usize) -> Self {
```

Comment:
Hmm,I think this is still `unsafe` since there's preconditions on calling it?

Reply:
What is the precondition?

Reply:
the set_fs thing that apparently I'm supposed to pretend doesn't exist :-)

Reply:
`set_fs()` is going away, yeah, but the concept of checking whether something is a valid userspace pointer is, as well? (i.e. `access_ok()`?)